models/webhook.go

```diff
@@ -798,3 +798,31 @@ func FindRepoUndeliveredHookTasks(repoID int64) ([]*HookTask, error) {
 	}
 	return tasks, nil
 }
+
+// DeleteDeliveredHookTasks deletes delivered hook tasks of one repository, leaving the most recent delivered based on the parameter
+func DeleteDeliveredHookTasks(repoID int64, numberDeliveriesToKeep int64) error {
+	var deliveryDates = make([]int64, 0, 10)
+	err := x.Table("hook_task").
+		Where("hook_task.repo_id = ? AND hook_task.is_delivered = ?", repoID, true).
+		Cols("hook_task.delivered").
+		Join("INNER", "repository", "hook_task.repo_id = repository.id").
+		And("repository.is_hook_task_purge_enabled = ?", true).
+		OrderBy("hook_task.delivered desc").
+		Limit(1, int(numberDeliveriesToKeep)).
+		Find(&deliveryDates)
+	if err != nil {
+		return err
+	}
+
+	if len(deliveryDates) > 0 {
+		deletes, err := x.
+			Where("repo_id = ? and is_delivered = ? and delivered <= ?", repoID, true, deliveryDates[0]).
```

Comment:
`deliveryDates[len(deliveryDates) - 1]`?

Reply:
I think its the same? Since the query to find the dates is limit 1, offset <num to keep>, there should only be one date in the array.